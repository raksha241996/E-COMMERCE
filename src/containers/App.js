import React from 'react';
import '../styles/containerStyles/App.scss'
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

import Home from './Home'
import Products from './Products'
import HomeIcon from '@material-ui/icons/Home'
import ShopIcon from '@material-ui/icons/Shop'
import ShoppingCartIcon from '@material-ui/icons/ShoppingCart'
import Cart from './Cart';

class App extends React.Component {
  constructor(props){
    super(props);

    this.state ={
      count : 0
    };

    this.handleCount = this.handleCount.bind(this);
  }

  componentWillUpdate(){
    parseInt(localStorage.getItem('count')) === 0 ? this.hideCount() : this.showCount();
  }

  componentDidMount(){
    this.setState({count : parseInt(localStorage.getItem('count')) })
  }

  handleCount(count){
    this.setState({count : count}); 
  }

  hideCount(){
    document.querySelector("#count").style.visibility = 'hidden';
  }
  showCount(){
    document.querySelector("#count").style.visibility = 'visible';
  }

  render() {
    return (
      
      <Router>
        <header>
          <nav className="topnav">
            <a href="#home" tabIndex="1" ><Link to="/"><HomeIcon/>Home</Link></a>
            <a href="#produts" tabIndex="1"> <Link to="/products/"><ShopIcon/>Products</Link></a>
            <a href="#cart"><Link to="/cart/"><ShoppingCartIcon/>Cart</Link> <span id="count">{ this.state.count }</span> </a>
          </nav>
        </header>
        <Route path="/" exact component={Home} />
        <Route path="/products/"  render={() =>  <Products handleCount = {this.handleCount} /> } />
        <Route path="/cart/" render={() =>  <Cart handleCount = {this.handleCount} />  } />
      </Router>
    
    );
  }
}

export default App;
