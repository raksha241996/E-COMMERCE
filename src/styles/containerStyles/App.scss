@import url('https://fonts.googleapis.com/css?family=Manjari&display=swap');

*{
  box-sizing: border-box;
}

.topnav {
  overflow: hidden;
  background-color: #333;
  display: flex;
}

.topnav a {
  float: left;
  color: #f2f2f2;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 21px;
  display: flex;
}

.topnav a:hover{
  background-color: #ddd;
  color: black;
}

.topnav a.active {
  background-color: lightseagreen;
  color: white;
}


.Home {
  position: relative;
  width: 100%;

  
img{

      width: 100%;
      height: auto;
  }

.btn{

      position: absolute;
      top: 90%;
      left: 50%;
      cursor: pointer;
      &:hover {
          color:white;
      }
      width:100;
      height: 100;
      border: 2px solid #4CAF50;
      border-radius: 10px;
      border-color: transparent;
      font-size: 16px;
      margin: 4px 2px;
  }
  
}

#count {
  border-radius : 50%;
  left : 50%;
  top : 50%;
  font-size: 1.5rem;
  height : 30px;
  width : 30px;
  font-size : 20px;
  background: white;
  color : black;
  visibility: visible; 
}


