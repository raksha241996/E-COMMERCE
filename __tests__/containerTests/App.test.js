import React from 'react';
import { shallow ,configure} from 'enzyme';
import App from '../../src/containers/App'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App component test with Enzyme', () => {
   it('renders without crashing', () => {
      const wrapper =   shallow(<App />);
         const instance = wrapper.instance()
         instance.hideCount()
         expect(wrapper).toMatchSnapshot();
    });
});


