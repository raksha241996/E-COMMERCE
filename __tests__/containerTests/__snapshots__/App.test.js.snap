// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
